qiskit_ibm_runtime/options/options.py

```diff
@@ -155,6 +169,56 @@ def _get_program_inputs(options: dict) -> dict:
                 inputs[key] = options[key]
         return inputs
 
+    @staticmethod
+    def _validate_program_inputs(
+        options: dict, primitive: PrimitiveType = "Sampler", backend: IBMBackend = None
+    ) -> None:
+        """Validate that program inputs (options) are valid
+        Raises:
+            ValueError: if optimization_level is out of the allowed range.
+            ValueError: if resilience_level is out of the allowed range.
+            ValueError: if resilience_level==3, backend is simulator and no coupling map
+        """
+        if not options.get("optimization_level") in list(
+            range(Options._MAX_OPTIMIZATION_LEVEL + 1)
+        ):
+            raise ValueError(
+                f"optimization_level can only take the values "
+                f"{list(range(Options._MAX_OPTIMIZATION_LEVEL + 1))}"
+            )
+
+        if primitive == "Estimator" and not options.get("resilience_level") in list(
+            range(Options._MAX_RESILIENCE_LEVEL_ESTIMATOR + 1)
+        ):
+            raise ValueError(
+                f"resilience_level can only take the values "
+                f"{list(range(Options._MAX_RESILIENCE_LEVEL_ESTIMATOR + 1))} in Estimator"
+            )
+        if (
+            primitive == "Sampler"
+            and options.get("resilience_level")
+            and not options.get("resilience_level") in [0, 1]
+        ):
+            raise ValueError(
+                f"resilience level can only take the values "
+                f"{list(range(Options._MAX_RESILIENCE_LEVEL_SAMPLER + 1))} in Sampler"
+            )
+        if (
+            options.get("resilience_level")
+            and options.get("resilience_level") == 3
```

Comment:
The first statement (line 207) seems unnecessary as the second one would fail if it's None.